app/views/components/_miller-columns-list.html.erb

```diff
@@ -3,7 +3,14 @@
     <li>
       <div class="govuk-checkboxes__item">
         <input class="govuk-checkboxes__input" type="checkbox" name="<%= name %>" value="<%= item[:value] %>" id="<%= id %>-<%= index %>"
-          <% if item[:checked].present? && item[:checked] %> checked <% end %>
+          <% if item[:checked].present? &&
+                  item[:checked] ||
+                    item[:items].any? do |child|
+                      child[:checked] ||
+                      child[:items].any? do |grandchild|
+                        grandchild[:checked]
+                      end
+                    end %> checked <% end %>
```

Comment:
It'd be nice if this logic existed in a helper method, rather than directly in the view – although I'm not sure there's an obvious place for it short of creating a new Miller Columns helper.

Reply:
Also: is this fixed regression covered by a test? If not, does it need to be? (Maybe not – just asking!)

Reply:
Okay cool i'll add a miller columns helper since it's the place that makes most sense to me.

It's actually covered in a controller spec https://github.com/alphagov/whitehall/blob/main/test/functional/admin/edition_world_tags_controller_test.rb#L50-L57

```
  view_test "should check a child taxon and its parents when only a child taxon is selected" do
    stub_publishing_api_links_with_taxons(@edition.content_id, [world_grandchild_taxon_content_id])

    get :edit, params: { edition_id: @edition }

    assert_select "input[value='#{world_child_taxon_content_id}'][checked='checked']"
    assert_select "input[value='#{world_grandchild_taxon_content_id}'][checked='checked']"
  end

```